components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -874,7 +908,7 @@ private void createTenant(String domain, String organizationId, String orgCreato
                     createTenantInfoBean(domain, organizationId, orgCreatorID, orgCreatorName, orgCreatorEmail));
         } catch (TenantMgtException e) {
             // Rollback created organization.
-            deleteOrganization(domain);
+            deleteOrganization(organizationId);
```

Comment:
When deleting orgnization it should pass the organizationId but passing domain could arise issues as domain can be not equals to organizatioId for 1st level organizations. (ex - IDAAS deployments)